### ISSUE TITLE
linter: false positive for module with import/extensions rule

### ISSUE THREAD

**Issue by james-camilleri** [oxc-project/oxc#11881]
### What version of Oxlint are you using?

1.2.0

### What command did you run?

pnpm oxlint

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "categories": {
    "correctness": "off"
  },
  "plugins": ["import"],
  "rules": {
    "import/extensions": [
      "error",
      "always",
      {
        "ignorePackages": true
      }
    ]
  }
}
```

### What happened?

There seems to be a false positive on the `import/extensions` rule for node modules using an export map, e.g. the `vitest` package with its `vitest/config` export. I've prepared a minimum reproduction here: https://github.com/james-camilleri/oxlint-extensions-reproduction.

Running ESlint  on a file with the following line (correctly) indicates no error, while running oxlint returns a false positive – the import statement shouldn't have an extension as it's a module.

```js
import { defineConfig } from "vitest/config";
```

I suspect the error is caused by this line:
https://github.com/oxc-project/oxc/blob/9a2548ae9dd9d218a60673db87ff139316d0525c/crates/oxc_linter/src/rules/import/extensions.rs#L279 
It seems to assume that anything with a `/` in it is not a module. I have absolutely no experience with rust however so this might be completely wrong.

**Comment by Hideman42** [comment]
I have the same issue here. It works fine locally but in a CI/CD GHA it goes over node_modules for no reasons...
I have to use the following in config `"ignorePatterns": ["**/node_modules", "**/dist", "**/output"],` but additionally this property is not inherited with `"extends"` property 😥

Would appreciate a fix for this. And actually, why node_modules even considered when linting? 🤔

**Comment by connorshea** [comment]
I created a repo based on yours with a few other oddities that exist in the current behavior of this rule: https://github.com/connorshea/oxlint-extensions-reproduction

**Comment by connorshea** [comment]
Based on my testing, this PR would fix the issue you appear to be seeing, at least partially: https://github.com/oxc-project/oxc/pull/14602

**Comment by sebastienbarre** [comment]
Why close it when the configuration is so obscure that you can't make it work. I can't figure out if my configuration is wrong, or if the code is not working.

**Comment by connorshea** [comment]
> Why close it when the configuration is so obscure that you can't make it work. I can't figure out if my configuration is wrong, or if the code is not working.

There are much more constructive ways of phrasing this :) Also the fix hasn't been released yet, it'll be released on Monday.

The configuration structure is exactly the same as the original import/extensions rule: https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/extensions.md

I'll work on improving the docs relatively soon, part of the problem is that we don't support auto-doc generation for rules that take tuples yet (see #16555).

**Comment by sebastienbarre** [comment]
My bad @connorshea, I got a notification and the next day an update on both the VS Code extension and the NPM packages, so I thought it was part of it.
